pkg/nsx/services/subnetport/store.go

```diff
@@ -0,0 +1,109 @@
+package subnetport
+
+import (
+	"errors"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+)
+
+// keyFunc is used to get the key of a resource, usually, which is the ID of the resource
+func keyFunc(obj interface{}) (string, error) {
+	switch v := obj.(type) {
+	case model.SegmentPort:
+		return *v.Id, nil
+	case types.UID:
+		return string(v), nil
+	default:
+		return "", errors.New("keyFunc doesn't support unknown type")
+	}
+}
+
+// indexFunc is used to get index of a resource, usually, which is the UID of the CR controller reconciles,
+// index is used to filter out resources which are related to the CR
+func indexFunc(obj interface{}) ([]string, error) {
+	res := make([]string, 0, 5)
+	switch o := obj.(type) {
+	case model.SegmentPort:
+		return filterTag(o.Tags), nil
+	default:
+		return res, errors.New("indexFunc doesn't support unknown type")
+	}
+}
+
+var filterTag = func(v []model.Tag) []string {
+	res := make([]string, 0, 5)
```

Comment:
Do we need to make a slice with capacity==5 here? I thinks we can assert that `len(res) == 1`?